deepdiver1975/tarstreamer/src/TarStreamer.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+namespace DeepDiver1975\TarStreamer;
+
+class TarStreamer {
+
+	const REGTYPE = 0;
+	const DIRTYPE = 5;
+	const XHDTYPE = 'x';
+
+	/**
+	 * Process in 1 MB chunks
+	 */
+	protected $blockSize = 1 * 1024 * 1024;
+	protected $outStream;
+	protected $needHeaders = false;
+
+	/**
+	 * Create a new TarStreamer object.
+	 *
+	 * @param array $options
+	 */
+	public function __construct($options = []){
+		if (isset($options['outstream'])){
+			$this->outstream = $options['outstream'];
+		} else {
+			$this->outstream = fopen('php://output', 'w');
+			// turn off output buffering
+			while (ob_get_level() > 0){
+				ob_end_flush();
+			}
+		}
+	}
+
+	/**
+	 * Send appropriate http headers before streaming the tar file and disable output buffering.
+	 * This method, if used, has to be called before adding anything to the tar file.
+	 *
+	 * @param string $archiveName Filename of archive to be created (optional, default 'archive.tar')
+	 * @param string $contentType Content mime type to be set (optional, default 'application/x-tar')
+	 */
+	public function sendHeaders($archiveName = 'archive.tar', $contentType = 'application/x-tar'){
+		if (headers_sent($headerFile, $headerLine)){
+			die(
+					"<p><strong>Error:</strong> Unable to send file " .
+					"$archiveName. HTML Headers have already been sent from " .
+					"<strong>$headerFile</strong> in line <strong>$headerLine" .
+					"</strong></p>"
+			);
+		}
+		$buffer = ob_get_contents();
+		if (!empty($buffer)){
+			die(
+					"\n<p><strong>Error:</strong> Unable to send file " .
+					"<strong>$archiveName.epub</strong>. Output buffer " .
```

Comment:
@LukasReschke the latter question goes to @McNetic as I was lazy enough to write this message on my own :)
https://github.com/McNetic/PHPZipStreamer/blob/master/src/ZipStreamer.php#L161